web/src/components/Sidebar.tsx

```diff
@@ -1,124 +1,60 @@
-  background-color: ${props => props.theme.legacy.colors.backgroundAccentColor};
-  min-height: ${props => props.headerHeight}px;
-  box-sizing: border-box;
-  padding: 8px;
-`
-
-const ActionBar = styled('nav')`
-  display: flex;
-  align-items: center;
-`
-
-const Content = styled('div')<{ headerHeight: number }>`
-  top: ${props => props.headerHeight}px;
-  height: calc(100% - ${props => props.headerHeight}px);
-  overflow: hidden auto;
-  padding: 0 32px;
-`
-
-const StyledIconButton = styled(IconButton)`
-  right: 4px;
+const Header = styled(Stack)`
+  align-items: flex-end;
+  justify-content: center;
+  padding: 0 8px;
 `
 
 type SidebarProps = {
   children: ReactNode
-  show: boolean
-  setShow: (show: boolean) => void
+  open: boolean
+  setOpen: (show: boolean) => void
   Footer?: ReactNode
   OpenButton?: ReactElement
 }
 
-const Sidebar = ({ children, show, setShow, Footer, OpenButton }: SidebarProps): ReactElement | null => {
-  useLockedBody(show)
+const Sidebar = ({ children, open, setOpen, Footer, OpenButton }: SidebarProps): ReactElement | null => {
   const { t } = useTranslation('layout')
-  const [scrollY, setScrollY] = useState<number>(0)
   const { headerHeight } = useWindowDimensions()
 
-  useLayoutEffect(() => {
-    if (show) {
-      setScrollY(window.scrollY)
-    }
-  }, [show])
+  // This is necessary to ensure the theme is correctly applied to the drawer content
+  const drawerContainer = document.getElementById(LAYOUT_ELEMENT_ID)
 
   return (
     <>
       {OpenButton ?? (
-        <ToggleContainer>
-          <IconButton onClick={() => setShow(!show)} aria-label={t('sideBarOpenAriaLabel')} aria-expanded={show}>
-            <MoreVertIcon />
-          </IconButton>
-        </ToggleContainer>
+        <IconButton onClick={() => setOpen(true)} aria-label={t('sideBarOpenAriaLabel')} aria-expanded={open}>
+          <MoreVertIcon />
+        </IconButton>
       )}
-      <Portal
-        className='sidebar'
-        show={show}
-        style={{
-          display: show ? 'block' : 'none',
-          top: scrollY > 0 ? `${scrollY}px` : undefined,
-        }}>
-        {/* disabled because this is an overlay for backdrop close */}
-        {/* eslint-disable-next-line styled-components-a11y/no-static-element-interactions,styled-components-a11y/click-events-have-key-events */}
-        <Overlay onClick={() => setShow(false)} show={show} />
-        <SidebarContainer>
-          <Heading headerHeight={headerHeight}>
-            <ActionBar>
-              <StyledIconButton onClick={() => setShow(false)} aria-label={t('sideBarCloseAriaLabel')}>
-                <CloseIcon />
-              </StyledIconButton>
-            </ActionBar>
-          </Heading>
-          <Content headerHeight={headerHeight}>{children}</Content>
-          {Footer}
-        </SidebarContainer>
-      </Portal>
+      <Drawer open={open} onClose={() => setOpen(false)} container={drawerContainer} anchor='right'>
```

Comment:
💭 Maybe a comment here that the `anchor='right'` is flipped automatically? Then again, might not be necessary. I'm not sure 😅

Reply:
Since mui takes care of a lot of direction specific things, I think this is to be expected and we would need to add this kind of comment in a few places. I'd therefore not put one.